### PR TITLE
i529: add --console command line option

### LINKED ISSUE #529: Add --console to add console logging by using ConsoleHandler
**Is your feature request related to a problem?** 

For the server and clients there is no way to write all log entries to
the console.

**Feature Description**: 

Add a --console option to add console logging.   When --console
used will output to stderr all log messages

**Have you considered other ways to accomplish the same thing?** 

Yes, there are none.

**Do you have any specific suggestions for how your feature would be ***implemented*** in PC^2?**

1.  Add a new constant in AppConstants
    public static final String CONSOLE_OPTION_STRING = "--console";

2. In InternalController.startMainUI just after the startLog add the following 3 lines:

if (parseArguments.isOptPresent(AppConstants.CONSOLE_OPTION_STRING)) {
   addConsoleLogging();
}

3.  add [--console] to the usage

**Additional context**:

### PR DESCRIPTION
**Pull request by lane55**
### Description of what the PR does

--console will output all log output to console

### Issue which the PR fixes

#529 

### Environment in which the PR was developed (OS,IDE, Java version, etc.)

Microsoft Windows [Version 10.0.19044.2130]

java version "1.8.0_121"
Java(TM) SE Runtime Environment (build 1.8.0_121-b13)
Java HotSpot(TM) 64-Bit Server VM (build 25.121-b13, mixed mode)

### Precise steps for _testing_ the PR (i.e., how to demonstrate that it works correctly)

Use --console on command line

All logging output will be written to console/stdout

### CODE REVIEW

File: src/edu/csus/ecs/pc2/AppConstants.java [hunk 1]
```diff
@@ -36,6 +36,8 @@ public class AppConstants {
     public static final String NO_SAVE_OPTION_STRING = "--nosave";
 
     public static final String PASSWORD_OPTION_STRING = "--password";
+
```
Comment (johnbrvc) [inline src/edu/csus/ecs/pc2/AppConstants.java:39]: Can we move this new option so it keeps the list of options alphabetical?  Right before --contestpassword.

**Review by johnbrvc — CHANGES_REQUESTED** [review]
Request to move new option so it appears alphabetically in AppConstants.java.

File: src/edu/csus/ecs/pc2/AppConstants.java (hunk 1 above)
Comment (lane55) [reply src/edu/csus/ecs/pc2/AppConstants.java:41]: I do not think that such a change is required.    With the invention of searching the code
finding constants does not require any particular order.    I do not see any significant value in alphabetizing
constants and think that is extra unnecessary work.   It also may require future, again, unnecessary merge work.

Comment (lane55) [reply src/edu/csus/ecs/pc2/AppConstants.java:41]: pushed fix.

**Review by johnbrvc — APPROVED** [review]
Not that it *really* matters as you indicated, but, I'm not sure why you moved it to where you did.  It is still not alphabetically correct; it should be before --contestpassword, but, it's no big deal.  Maybe you were putting it alphabetically before CONTEST_ID_OPTION_STRING? But none of the other constant variable names are alphabetical.  Not worth holding up the approval for.

**Comment by troy2914** [comment]
Please rebase this branch with the current develop.  I am unable to build this with gh pr checkout 580, as it reverts build.xml to an version with broken bootlcasspath references.

**Comment by troy2914** [comment]
> Please rebase this branch with the current develop. I am unable to build this with gh pr checkout 580, as it reverts build.xml to an version with broken bootlcasspath references.

I have copied the current createVERSIONandJar.xml to this branch to build it,  but trying it on the pc2admin and pc2server it does not like --console and reports this error:
```
java.lang.IllegalArgumentException: invalid option '--console'
	at edu.csus.ecs.pc2.core.ParseArguments.loadArgs(ParseArguments.java:352)
	at edu.csus.ecs.pc2.core.ParseArguments.<init>(ParseArguments.java:557)
	at edu.csus.ecs.pc2.core.InternalController.start(InternalController.java:2968)
	at edu.csus.ecs.pc2.Starter.main(Starter.java:50)
```

**Comment by troy2914** [comment]
> > Please rebase this branch with the current develop. I am unable to build this with gh pr checkout 580, as it reverts build.xml to an version with broken bootlcasspath references.
> 
> I have copied the current createVERSIONandJar.xml to this branch to build it, but trying it on the pc2admin and pc2server it does not like --console and reports this error:
> 
> ```
> java.lang.IllegalArgumentException: invalid option '--console'
> 	at edu.csus.ecs.pc2.core.ParseArguments.loadArgs(ParseArguments.java:352)
> 	at edu.csus.ecs.pc2.core.ParseArguments.<init>(ParseArguments.java:557)
> 	at edu.csus.ecs.pc2.core.InternalController.start(InternalController.java:2968)
> 	at edu.csus.ecs.pc2.Starter.main(Starter.java:50)
> ```

Talking with JohnB it sounds like with #524 the CONSOLE_OPTION_STRING needs to be added to AppConstants.allOptionStrings list.

**Review by troy2914 — CHANGES_REQUESTED** [review]
CONSOLE_OPTION_STRING needs to be added to allOptionStrings, otherwise you wind up with 
```
java.lang.IllegalArgumentException: invalid option '--console'
	at edu.csus.ecs.pc2.core.ParseArguments.loadArgs(ParseArguments.java:352)
	at edu.csus.ecs.pc2.core.ParseArguments.<init>(ParseArguments.java:557)
	at edu.csus.ecs.pc2.core.InternalController.start(InternalController.java:2968)
	at edu.csus.ecs.pc2.Starter.main(Starter.java:50)
```
and a PC2 Halted OK screen.

**Comment by lane55** [comment]
merged develop into branch.

**Comment by lane55** [comment]
pushed fix for 
| java.lang.IllegalArgumentException: invalid option '--console'

**Review by johnbrvc — APPROVED** [review]
Retested.  Seems to work.

**Review by troy2914 — APPROVED** [review]
--console works on server and admin,  didn't test others.

**Pull request merged**